sdk/core/System.ClientModel/gen/ModelReaderWriterContextGenerator.Emitter.cs

```diff
@@ -51,6 +52,18 @@ private void EmitContextClass(ModelReaderWriterContextGenerationSpec contextGene
             builder.AppendLine(indent, $"namespace {contextGenerationSpec.Type.Namespace};");
             builder.AppendLine();
 
+            // Add pragma directives for each experimental type
```

Comment:
I would do this inline for each model that needs it.  It will make the code simpler and more efficient.